vector/src/main/java/im/vector/app/features/home/HomeActivityViewModel.kt

```diff
@@ -130,6 +133,47 @@ class HomeActivityViewModel @AssistedInject constructor(
                 .launchIn(viewModelScope)
     }
 
+    /**
+     * Handle threads migration. The migration includes:
+     * - Notify users that had io.element.thread enabled from labs
+     * - Re-Enable m.thread to those users (that they had enabled labs threads)
+     * - Handle migration when threads are enabled by default
+     */
+    private fun initThreadsMigration() {
+        // Notify users
+        if (vectorPreferences.shouldNotifyUserAboutThreads() && vectorPreferences.areThreadMessagesEnabled()) {
```

Comment:
if each block has a return we could use a `when` and avoid multiple exit points 

```kotlin
when {
 vectorPreferences.shouldNotifyUserAboutThreads() && vectorPreferences.areThreadMessagesEnabled() -> ...
 vectorPreferences.shouldNotifyUserAboutThreads() && vectorPreferences.shouldMigrateThreads() -> ...
}
```

Reply:
I agree 100%